openwisp_monitoring/device/admin.py

```diff
@@ -160,56 +167,68 @@ def get_form(self, request, obj=None, **kwargs):
         return super().get_form(request, obj, **kwargs)
 
     def get_fields(self, request, obj=None):
-        fields = super().get_fields(request, obj)
+        fields = list(super().get_fields(request, obj))
         if not obj or obj.monitoring.status in ['ok', 'unknown']:
             return fields
-        fields = list(fields)
+        fields.insert(fields.index('last_ip'), 'health_status')
         fields.insert(fields.index('health_status') + 1, 'health_checks')
         return fields
 
     def get_readonly_fields(self, request, obj=None):
-        readonly_fields = super().get_readonly_fields(request, obj)
+        readonly_fields = list(super().get_readonly_fields(request, obj))
+        readonly_fields.append('health_status')
         if not obj or obj.monitoring.status in ['ok', 'unknown']:
             return readonly_fields
-        readonly_fields = list(readonly_fields)
         readonly_fields.append('health_checks')
         return readonly_fields
 
+    def get_list_display(self, request):
+        list_display = list(super().get_list_display(request))
+        list_display.insert(list_display.index('config_status'), 'health_status')
+        return list_display
 
-def device_admin_get_inlines(self, request, obj):
-    # copy the list to avoid modifying the original data structure
-    inlines = list(super(DeviceAdmin, self).get_inlines(request, obj))
-    if not obj or obj._state.adding:
-        inlines.remove(MetricInline)
-        return inlines
-    return inlines
-
-
-DeviceAdmin.inlines += [CheckInline, MetricInline]
-# This attribute needs to be set for nested inline
-for inline in DeviceAdmin.inlines:
-    if not hasattr(inline, 'sortable_options'):
-        inline.sortable_options = {'disabled': True}
+    def get_list_select_related(self, request):
+        list_select_related = list(super().get_list_select_related(request))
+        list_select_related.append('monitoring')
+        return list_select_related
 
-DeviceAdmin.get_inlines = device_admin_get_inlines
+    def get_list_filter(self, request):
+        list_filter = list(super().get_list_filter(request))
+        list_filter.insert(
+            0, 'monitoring__status',
+        )
+        return list_filter
```

Comment:
why do you override the method instead of just doing:

```python
list_filter = ['monitoring__status'] + BaseDeviceAdmin.list_filter
```

Let's aim for less code.